webrtc/README.md

```diff
@@ -114,7 +114,9 @@ Safari, Edge, Firefox)._
 #### Browser to browser or non-public server node
 
 - Can _Browser_ know upfront its UDP port which it is listening for incoming
-  connections on?
+  connections on? Does the browser reuse the UDP port across many WebRTC
```

Comment:
No, browsers cannot determine upfront what UDP port it will be using for a new RTCPeerConnection. Every new connection will use a different UDP port as well. 
Furthermore, you cannot add your own candidates for your own local description, which means that you cannot set your own port on the browser side. We have tried modifying our browsers SDP packet's candidates and when you setLocalDescription on that the browser errors and gives you the following error message: `Uncaught (in promise) DOMException: Adding your own candidate attributes is not supported`

Reply:
Thanks @ec2! While I don't see this hurting our browser-to-server story, as far as I can tell this requires us to rethink our browser-to-browser story. More specifically I don't see how we can do browser-to-browser without doing STUN but libp2p-identify to some other public WebRTC server.

Reply:
Browser-to-browser will definitely look different from browser-to-server. What I am envisioning is what we may have talked about before. Namely, using a mutually connected server (if not directly connected then through a DHT perhaps?) to relay an SDP packet exchange.

Reply:
We would also need a separate signaling mechanism (DHT or mutually connected server/ relay) for sending ICE trickle candidates. If the browser adds only host candidates to the SDP then browsers behind different NATs will be unable to connect. Also, a STUN server is required for both browsers to discover their server-reflexive candidates (these are publicly reachable and create a binding on the local NAT which allows UDP packets into the subnet), but these can be separate and the browsers do not need to coordinate.

Reply:
Regarding trickle ICE: A trick that's often used is to just wait for a bit to gather candidates before sending exchanging SDP. E.g.:
https://github.com/pion/webrtc/blob/c1467e4871c78ee3f463b50d858d13dc6f2874a4/examples/insertable-streams/main.go#L141-L142

Reply:
Thanks @backkem for the link. Documented for now via 8208f02. I will focus the first iteration of this specification on browser-to-server, thus delaying this conversation to the second iteration and resolving here for now.